src/events/embedded/types.ts

```diff
@@ -1,116 +1,81 @@
-export interface IEmbeddedMessageElementsButton {
+interface Session {
   id: string;
-  title: string;
-  action?: IEmbeddedMessageElementsButtonAction;
+  start?: number;
+  end?: number;
 }
-export interface IEmbeddedMessageElementsButtonAction {
+
+interface Impression {
+  messageId: string;
+  displayCount: number;
+  displayDuration: number;
+}
+
+interface ButtonAction {
   type: string;
   data?: string;
 }
-export interface IEmbeddedMessageElements {
+
+interface Button {
+  id: string;
+  title: string;
+  action?: ButtonAction;
```

Comment:
Is the action prop ever expected to not exist a button? If a button is defined, wouldn't it have either an openUrl or customAction action?

Reply:
it's optional so it is possible